management/src/main/java/io/micronaut/management/endpoint/env/EnvironmentEndpoint.java

```diff
@@ -17,104 +17,125 @@
+)
 public class EnvironmentEndpoint {
 
     /**
      * Endpoint name.
      */
     public static final String NAME = "env";
 
-    public static final String[] PROPERTY_NAMES_TO_MASK = new String[] {
-            "password", "credential", "certificate", "key", "secret", "token"
-    };
+    /**
+     * Endpoint default enabled.
+     */
+    public static final boolean DEFAULT_ENABLED = false;
 
     private final Environment environment;
-    private final List<Pattern> maskPatterns;
+    private final EnvironmentEndpointFilter environmentFilter;
 
     /**
      * @param environment The {@link Environment}
      */
     public EnvironmentEndpoint(Environment environment) {
+        this(environment, null);
+    }
+
+    /**
+     * @param environment The {@link Environment}
+     * @param environmentFilter The registered {@link EnvironmentEndpointFilter} bean if one is registered
+     */
+    @Inject
+    public EnvironmentEndpoint(Environment environment, @Nullable EnvironmentEndpointFilter environmentFilter) {
         this.environment = environment;
-        this.maskPatterns = Arrays.stream(PROPERTY_NAMES_TO_MASK)
-                .map(s -> Pattern.compile(".*" + s + ".*", Pattern.CASE_INSENSITIVE))
-                .collect(Collectors.toList());
+        this.environmentFilter = environmentFilter;
     }
 
     /**
+     * @param principal The current {@link Principal} if one exists
      * @return The environment information as a map with the following keys: activeEnvironments, packages and
      * propertySources.
      */
     @Read
-    public Map<String, Object> getEnvironmentInfo() {
+    public Map<String, Object> getEnvironmentInfo(@Nullable Principal principal) {
+        EnvironmentFilterSpecification filter = createFilterSpecification(principal);
+
         Map<String, Object> result = new LinkedHashMap<>();
         result.put("activeEnvironments", environment.getActiveNames());
         result.put("packages", environment.getPackages());
         Collection<Map<String, Object>> propertySources = new ArrayList<>();
         environment.getPropertySources()
                 .stream()
                 .sorted(Comparator.comparing(PropertySource::getOrder))
-                .forEach(ps -> propertySources.add(buildPropertySourceInfo(ps)));
+                .map(ps -> buildPropertySourceInfo(ps, filter))
+                .forEach(propertySources::add);
         result.put("propertySources", propertySources);
         return result;
     }
 
     /**
      * @param propertySourceName The {@link PropertySource} name
+     * @param principal The current {@link Principal} if one exists
      * @return a map with all the properties defined in the property source if it exists; null otherwise.
      */
     @Read
-    public Map<String, Object> getProperties(@Selector String propertySourceName) {
+    public Map<String, Object> getProperties(@Selector String propertySourceName, @Nullable Principal principal) {
```

Comment:
This is breaking binary compatibility. Add another deprecated method with the original signature.